src/org/zaproxy/zap/extension/websocket/client/HttpHandshakeBuilder.java

```diff
@@ -73,9 +73,21 @@ public static HttpRequestHeader getHttpHandshakeRequestHeader(URI hostUri) throw
      * @param httpRequestHeader the http request header
      */
     public static void addGeneratedWebSocketKey(HttpRequestHeader httpRequestHeader){
+        httpRequestHeader.addHeader(SEC_WEB_SOCKET_KEY, null); //Remove if any
         httpRequestHeader.addHeader(SEC_WEB_SOCKET_KEY, WebSocketUtils.generateSecWebSocketKey());
     }
     
+    
+    /**
+     * Generates and add a Sec-WebSocket-key to {@code httpRequestHeader}
+     *
+     * @param httpRequestHeader the http request header
+     */
+    public static void addWebSocketKey(HttpRequestHeader httpRequestHeader, String websocketKey){
+        httpRequestHeader.addHeader(SEC_WEB_SOCKET_KEY, null); //Remove if any
+        httpRequestHeader.addHeader(SEC_WEB_SOCKET_KEY, websocketKey);
```

Comment:
Could call `setHeader` (instead of the two `addHeader`), same for the above method.